### ISSUE TITLE
internal error: cannot dynamically construct operator unknown

### ISSUE THREAD

**Issue by georgysavva** [cockroachdb/cockroach#47230]
**Describe the problem**

I've got this stacktrace:
```
pq: internal error: cannot dynamically construct operator unknown
DETAIL: stack trace:
/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:17375: DynamicConstruct()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/norm/custom_funcs.go:1351: NegateComparison()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go:8521: ConstructNot()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scalar.go:308: buildScalar()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/util.go:411: resolveAndBuildScalar()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:921: buildWhere()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:809: buildSelectClause()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:759: buildSelect()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:241: buildStmt()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:185: Build()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:419: buildExecMemo()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:161: makeOptimizerPlan()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:743: makeExecPlan()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:625: dispatchToExecutionEngine()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:416: execStmtInOpenState()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:98: execStmt()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1243: execCmd()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1172: run()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:444: ServeConn()
/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:584: func1()
runtime/asm_amd64.s:1337: goexit()
```
**To Reproduce**
I have this table:
```
create table if not exists tmp
(
        list_of_values _text,
        rowid int8 default unique_rowid() not null
                constraint "primary"
                        primary key
);
```
With the following data:
```
select * from tmp;
  list_of_values |       rowid         
+----------------+--------------------+
  {1,2}          | 544935729459068929  
  {1,4}          | 544964749078495233  
```
I wanted to select rows with list_of_values that doesn't share any item with array I provide. In other words, select rows that will have zero intersection with array I provide. The query that I tried to execute:
```
select * from tmp where not list_of_values && array['2','3'];
```
It failed and resulted in the crash that I attached above. Postgres supports that type of operation, I tested it. 
BTW the opposite operation (without `not`) works fine in CockroachDB:
```
select * from tmp where list_of_values && array['2','3'];
  list_of_values |       rowid         
+----------------+--------------------+
  {1,2}          | 544935729459068929  

```

**Environment:**
 - CockroachDB version 19.2.5 
 - Single node cluster 
 - Local MacOS
 - Client app: `cockroach sql`

**Comment by RaduBerinde** [comment]
This was fixed in #43228. Unfortunately it was only recently backported to the 19.2 branch (#46539). The fix will be part of 19.2.6.

**Comment by georgysavva** [comment]
Thanks!
